### PR TITLE
xero import settings ts settings

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts [hunk 1]
```diff
@@ -7,9 +19,177 @@ import { Component, OnInit } from '@angular/core';
...
+
+  isOnboarding: boolean;
+
+  importSettings: XeroImportSettingGet;
+
+  fyleExpenseFields: FyleField[];
+
+  workspaceGeneralSettings: XeroWorkspaceGeneralSetting;
+
+  customMappedFyleFields: any;
```
Comment (DhaaraniCIT) [inline src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts:32]: removed in the next PR

File: src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts [hunk 2]
```diff
@@ -7,9 +19,177 @@ import { Component, OnInit } from '@angular/core';
...
+
+  isProjectMapped: boolean;
+
+  taxCodes: DestinationAttribute[];
+
+  importSettingsForm: FormGroup<any>;
+
+  customFieldType: string;
+
+  customFieldControl: any;
```
Comment (DhaaraniCIT) [inline src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts:44]: fixed in next PR

File: src/app/core/models/xero/xero-configuration/xero-import-settings.model.ts [hunk 3]
```diff
@@ -56,6 +56,21 @@ export interface XeroImportSettingFormOption extends SelectFormOption {
...
 
 export class XeroImportSettingModel extends ImportSettingsModel {
+
+  static mapAPIResponseToFormGroup(importSettings: XeroImportSettingGet | null, xeroFields: IntegrationField[]): FormGroup {
+    const expenseFieldsArray = importSettings?.mapping_settings ? this.constructFormArray(importSettings.mapping_settings, xeroFields) : [];
+    return new FormGroup({
+      importCategories: new FormControl(importSettings?.workspace_general_settings.import_categories ?? false),
+      expenseFields: new FormArray(expenseFieldsArray),
+      chartOfAccountTypes: new FormControl(importSettings?.workspace_general_settings.charts_of_accounts ? importSettings.workspace_general_settings.charts_of_accounts : ['Expense']),
+      importCustomers: new FormControl(importSettings?.workspace_general_settings.import_customers ?? false),
```
Comment (ashwin1111) [inline src/app/core/models/xero/xero-configuration/xero-import-settings.model.ts:66]: not needed

File: src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts [hunk 4]
```diff
@@ -7,9 +19,177 @@ import { Component, OnInit } from '@angular/core';
...
+
+  fyleExpenseFields: FyleField[];
+
+  workspaceGeneralSettings: XeroWorkspaceGeneralSetting;
+
+  customMappedFyleFields: any;
+
+  xeroExpenseFields: IntegrationField[];
+
+  isProjectMapped: boolean;
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts:36]: not required

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**